### PR TITLE
-property that indicated when a device is ready to be used

### PR DESCRIPTION
**Pull request by raykamp**

### CODE REVIEW

File: lib/noble-device.js [hunk 1]
```diff
@@ -66,7 +68,10 @@ NobleDevice.prototype.connectAndSetUp = NobleDevice.prototype.connectAndSetup =
       return callback(error);
     }
 
-    this.discoverServicesAndCharacteristics(callback);
+    this.discoverServicesAndCharacteristics(function() {
+      this.connectedAndSetUp = true;
+      callback();
+    });
```
Comment (sandeepmistry) [inline lib/noble-device.js]: Do you think we need a ```}.bind(this));``` here?

**Comment by sandeepmistry** [comment]
@raykamp thanks! I'll see if the ```bind``` this is needed ...

**Comment by sandeepmistry** [comment]
@raykamp it was needed, the value remained ``false`` when I tested with ``node-flower-power``.

Here's the commit that bind's the this value: https://github.com/sandeepmistry/noble-device/commit/d1e789d9cb9510f4fcf46b6934df4c1f371a63dc

**Comment by raykamp** [comment]
good catch @sandeepmistry !

**Pull request merged**